aws/signer/v4/middleware.go

```diff
@@ -235,7 +247,9 @@ func (s *SignHTTPRequest) HandleFinalize(ctx context.Context, in middleware.Fina
 		return out, metadata, &SigningError{Err: fmt.Errorf("failed to retrieve credentials: %w", err)}
 	}
 
-	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime())
+	err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime(), func(options *SignerOptions) {
+		options.LogSigning = s.logSigning
+	})
```

Comment:
super minor, but you could avoid the additional function closure being allocated by creating a method on `SignHTTPRequest` that sets the logging, and can be passed in as a functional option.

```go
func (s *SignHTTPRequest) addLogSignerOptions(o *SignerOptions) {
     o.LogSigning = s.logSigning
)
```

```go
err = s.signer.SignHTTP(ctx, credentials, req.Request, payloadHash, signingName, signingRegion, sdk.NowTime(),
     s.addLogSignerOptions,
)
```